routers/repo/branch.go

```diff
@@ -52,7 +52,22 @@ func Branches(ctx *context.Context) {
 	ctx.Data["PageIsViewCode"] = true
 	ctx.Data["PageIsBranches"] = true
 
-	ctx.Data["Branches"] = loadBranches(ctx)
+	page := ctx.QueryInt("page")
+	if page <= 1 {
+		page = 1
+	}
+
+	pageSize := ctx.QueryInt("limit")
+	if pageSize <= 0 {
+		pageSize = git.BranchesRangeSize
```

Comment:
I think this belongs in `setting.UI` rather than in `git`.

Reply:
I think I'm fine with leaving it in git - else we should also move **CommitsRangeSize** to settings ...

Reply:
it's not used in any git module code and should be configurable.

Reply:
OK I've made it configurable.